lnrpc/walletrpc/walletkit_server.go

```diff
@@ -672,6 +676,64 @@ func (w *WalletKit) PublishTransaction(ctx context.Context,
 	return &PublishResponse{}, nil
 }
 
+// RemoveTransaction attempts to remove the transaction and all of its
+// descendants resulting from further spends of the outputs of the provided
+// transaction id.
+// NOTE: We do not remove the transaction from the rebroadcaster which might
+// run in the background rebroadcasting not yet confirmed transactions. We do
+// not have access to the rebroadcaster here nor should we. This command is not
+// a way to remove transactions from the network. It is a way to shortcircuit
+// wallet utxo housekeeping while transactions are still unconfirmed and we know
+// that a transaction will never confirm because a replacement already pays
+// higher fees.
+func (w *WalletKit) RemoveTransaction(_ context.Context,
+	req *GetTransactionRequest) (*RemoveTransactionResponse, error) {
+
+	// If the client doesn't specify a hash, then there's nothing to
+	// return.
+	if req.Txid == "" {
+		return nil, fmt.Errorf("must provide a transaction hash")
+	}
+
+	txHash, err := chainhash.NewHashFromStr(req.Txid)
+	if err != nil {
+		return nil, err
+	}
+
+	// Query the tx store of our internal wallet for the specified
+	// transaction.
+	res, err := w.cfg.Wallet.GetTransactionDetails(txHash)
+	if err != nil {
+		return nil, fmt.Errorf("transaction with txid=%v not found "+
+			"in the internal wallet store", txHash)
+	}
+
+	// Only allow unconfirmed transactions to be removed because as soon
+	// as a transaction is confirmed it will be evaluated by the wallet
+	// again and the wallet state would be updated in case the user had
+	// removed the transaction accidentally.
+	if res.NumConfirmations > 0 {
```

Comment:
We might think of a `--i_know_what_i_am_doing` flag and also allow the removal of confirmed txids ?

Reply:
See my comment above about doing it in a separate PR (if even...). Also, when it comes to naming, I think we'd want something more along the lines of `--danger-allow-confirmed` since apparently the "I know what I'm doing" approach doesn't seem to work for some users. Psychologically it seems people are more inclined to think they know what they're doing while if something is labelled as "dangerous" they might be more careful (even if they happen to actually know what they are doing).

Reply:
well said, I observed the same behavior :)